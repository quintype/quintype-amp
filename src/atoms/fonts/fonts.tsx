import React from "react";
import { Helmet } from "react-helmet";
import { withStoryAndConfig } from "../../context";

export const FontsBase = ({ config }) => {
  const { primary, secondary } = config.ampConfig.fonts;

  return (
    <Helmet>
      <link rel="preconnect dns-prefetch" href="https://fonts.gstatic.com" crossorigin="anonymous" />
      <link
        rel="stylesheet"
        crossorigin="anonymous"
        href={`https://fonts.googleapis.com/css?family=${primary.url}&display=swap`}
      />
      <link
        rel="stylesheet"
        crossorigin="anonymous"
        href={`https://fonts.googleapis.com/css?family=${secondary.url}&display=swap`}
      />
    </Helmet>
  );
};

export const Fonts = withStoryAndConfig(FontsBase);
