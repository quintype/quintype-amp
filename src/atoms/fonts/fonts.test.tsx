import { shallow } from "enzyme";
import { FontsBase } from "./fonts";
import React from "react";

const dummyConfig = {
  ampConfig: {
    fonts: {
      primary: {
        url: "Mukta+Malar:300,400,600,700",
        family: '"Mukta Malar", sans-serif'
      },
      secondary: {
        url: "Mukta+Malar:400,400italic,700,700italic",
        family: '"Mukta Malar",sans-serif'
      }
    }
  }
};

describe("Fonts component", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<FontsBase config={dummyConfig} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("Should preconnect to fonts provider and preload fonts", () => {
    const wrapper = shallow(<FontsBase config={dummyConfig} />);
    expect(
      wrapper
        .find(`link`)
        .at(0)
        .prop("rel")
    ).toBe("preconnect dns-prefetch");
    expect(
      wrapper
        .find(`link`)
        .at(0)
        .prop("crossorigin")
    ).toBe("anonymous");
    expect(
      wrapper
        .find(`link`)
        .at(1)
        .prop("rel")
    ).toBe("preload");
    expect(
      wrapper
        .find(`link`)
        .at(1)
        .prop("crossorigin")
    ).toBe("anonymous");
    expect(
      wrapper
        .find(`link`)
        .at(1)
        .prop("href")
    ).toBe("https://fonts.googleapis.com/css?family=Mukta+Malar:300,400,600,700");
    expect(
      wrapper
        .find(`link`)
        .at(2)
        .prop("rel")
    ).toBe("preload");
    expect(
      wrapper
        .find(`link`)
        .at(2)
        .prop("crossorigin")
    ).toBe("anonymous");
    expect(
      wrapper
        .find(`link`)
        .at(2)
        .prop("href")
    ).toBe("https://fonts.googleapis.com/css?family=Mukta+Malar:400,400italic,700,700italic");
  });
});
