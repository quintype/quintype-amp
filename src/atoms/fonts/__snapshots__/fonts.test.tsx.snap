// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fonts component Should match snapshot 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <link
    crossorigin="anonymous"
    href="https://fonts.gstatic.com"
    rel="preconnect dns-prefetch"
  />
  <link
    as="style"
    crossorigin="anonymous"
    href="https://fonts.googleapis.com/css?family=Mukta+Malar:300,400,600,700&display=swap"
    rel="preload"
  />
  <link
    as="style"
    crossorigin="anonymous"
    href="https://fonts.googleapis.com/css?family=Mukta+Malar:400,400italic,700,700italic&display=swap"
    rel="preload"
  />
  <link
    as="style"
    crossorigin="anonymous"
    href={null}
    rel="preload"
  />
</HelmetWrapper>
`;
